### ISSUE TITLE
StackOverflowError while large project opening

### ISSUE THREAD

**Issue by ArtsiomCh** [manifold-systems/manifold#41]
**Describe the bug**
StackOverflowError while large project opening

**To Reproduce**
Open large project: [IntellyJ Idea CE source code](https://github.com/JetBrains/intellij-community)

**Desktop (please complete the following information):**
IntelliJ IDEA 2018.3.2 (Community Edition)
Build #IC-183.4886.37, built on December 17, 2018
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-141-generic
- Manifold version: 0.39-alpha
- Manifold IntelliJ plugin version: v0.39-alpha

**Stack trace**
```
java.lang.StackOverflowError
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:136)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:121)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
        .... same line till the end of the stack
```

**Comment by rsmckinney** [comment]
in progress

**Comment by rsmckinney** [comment]
This is fixed with plugin v0.40-alpha which is pending release, probably another couple of days before JetBrains verifies

**Comment by ArtsiomCh** [comment]
Unfortunately still have `StackOverflowError`... 
Manifold version: 0.40-alpha
Manifold IntelliJ plugin version: v0.40-alpha
Exceptions in reverse order:
```
java.lang.StackOverflowError
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:116)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:97)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
...
```
```
java.lang.StackOverflowError
	at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
	at manifold.api.fs.cache.PathCache.getFqnForFile(PathCache.java:109)
	at manifold.ext.ExtensionManifold.handlesFile(ExtensionManifold.java:102)
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:136)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:121)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
...
```
```
java.lang.StackOverflowError
	at manifold.util.cache.FqnCache.getNode(FqnCache.java:60)
	at manifold.util.cache.FqnCache.get(FqnCache.java:74)
	at manifold.api.type.ResourceFileTypeManifold.findTopLevelFqn(ResourceFileTypeManifold.java:325)
	at manifold.api.type.ResourceFileTypeManifold.isType(ResourceFileTypeManifold.java:297)
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:116)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:97)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
...
```
```
java.lang.StackOverflowError
	at manifold.api.type.ResourceFileTypeManifold.isType(ResourceFileTypeManifold.java:297)
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:116)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:97)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:107)
...
```
```
java.lang.StackOverflowError
	at com.intellij.openapi.vfs.VirtualFile.getExtension(VirtualFile.java:175)
	at manifold.ij.fs.IjFile.getExtension(IjFile.java:100)
	at manifold.api.type.ResourceFileTypeManifold.handlesFile(ResourceFileTypeManifold.java:253)
	at manifold.api.host.IModule.findTypeManifoldsFor(IModule.java:136)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:121)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
	at manifold.ij.core.ManModule.findTypeManifoldsFor(ManModule.java:131)
...
```

**Comment by rsmckinney** [comment]
The stack traces look like they are from older versions of both Manifold and the Manifold plugin.  Be sure you are using the latest plugin release 0.40-alpha

**Comment by ArtsiomCh** [comment]
Hmm... looks like evrething is updated:
![image](https://user-images.githubusercontent.com/14268320/52149570-22c87800-263b-11e9-96bd-6db8ce416b70.png)
![image](https://user-images.githubusercontent.com/14268320/52149617-50152600-263b-11e9-84bb-a450a6a4f101.png)
jar was taken from here: http://manifold.systems/docs.html#binaries

**Comment by rsmckinney** [comment]
![image](https://user-images.githubusercontent.com/588843/52153505-29e98900-262f-11e9-9872-6f026ae7c20a.png)


Yeah. . .  the manifold binaries I bundled with the plugin are an older snapshot version.  I'll have a new plugin release ready later today.  Sorry about that.

**Comment by rsmckinney** [comment]
Fixed with 0.41-alpha, submitted to JetBrains repo.  May take a couple of days before it's available, you can download it directly at https://plugins.jetbrains.com/plugin/10057-manifold.

**Comment by ArtsiomCh** [comment]
> The plugin update is pending JetBrains approval

Probably (by my experience), it gonna last till Monday at least.
Any way to just download jar file?...

**Comment by rsmckinney** [comment]
I've just published another release (0.42-alpha) along with a plugin release (the plugin and manifold always have matching release numbers).

You can download the plugin on the [JetBrains page](https://plugins.jetbrains.com/plugin/10057-manifold).  You can see the list of releases near the bottom of the page.  Here's a [direct link](https://plugins.jetbrains.com/plugin/download?rel=true&updateId=57690)

**Comment by ArtsiomCh** [comment]
Thank you for the direct link!
Unfortunately, no one except owner can see not approved versions (fall into that issue before with my plugins too). That's what I see on your plugin page:
![image](https://user-images.githubusercontent.com/14268320/52169660-3a226680-270a-11e9-83a2-fd6151454ce8.png)
